File: 2545. Sort the Students by Their Kth Score/sort-the-students-by-their-kth-score.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution
{
public:
    // Function to sort the students' scores based on a specific index
    // Input: score - a vector of vectors containing the students' scores
    //        k - the index based on which the sorting should be performed
    // Output: a sorted vector of vectors containing the students' scores
    std::vector<std::vector<int>> sortTheStudents(std::vector<std::vector<int>> &score, int k)
    {
        // Sort the score vector based on the value at index 'k' in descending order
        std::sort(score.begin(), score.end(), [k](const std::vector<int> &a, const std::vector<int> &b)
                  {
                      return a[k] > b[k]; // Return true if value at index 'k' in vector 'a' is greater than value at index 'k' in vector 'b'
                  });
        return score; // Return the sorted score vector
    };
};

File: 2545. Sort the Students by Their Kth Score/sort-the-students-by-their-kth-score.java
import java.util.Arrays;

public class Solution {
    
    // Method to sort the students' scores based on a specific index
    // Input: score - a 2D array containing the students' scores
    //        k - the index based on which the sorting should be performed
    // Output: a sorted 2D array containing the students' scores
    public int[][] sortTheStudents(int[][] score, int k) {
        // Sort the 'score' array based on the value at index 'k' in descending order
        Arrays.sort(score, (a, b) -> Integer.compare(b[k], a[k]));
        return score; // Return the sorted 2D array
    }
}
